sccp/figures/figureThomsonDrop.py

```diff
@@ -1,41 +1,202 @@
+        "VPREB3",
     ]
+    NaiveBCellGeneSet = marker_genes_2["B Naive"]
+    NKCellGeneSet = np.unique(marker_genes_1["Natural Killers"] + marker_genes_2["NK"])
+    DCCellGeneSet = np.unique(
+        marker_genes_1["Dendritic cells"] + marker_genes_2["cDCs"]
+    )
+    pDCGeneSet = marker_genes_2["pDCs"]
+    TCellGeneSet = np.unique(marker_genes_1["T cells"] + marker_genes_2["T Cells"])
+    CytoTCellGeneSet = marker_genes_2["Cytotoxic T"]
+    memorytcellgeneset = marker_genes_2["Memory T"]
+    cDCgeneset = marker_genes_2["cDCs"]
 
-    ax, f = getSetup((12, 12), (2, 3))
+    plotDifferentialExpression(data, "CTRL4", "B Cells", bCellGeneSet, rank, *ax[0:4])
+    plotDifferentialExpression(
+        data,
+        "CTRL4",
+        "Naive B Cells",
+        NaiveBCellGeneSet,
+        rank,
+        *ax[4:8],
+        ct2=True,
+    )
+    plotDifferentialExpression(data, "CTRL4", "NK Cells", NKCellGeneSet, rank, *ax[8:12])
+    plotDifferentialExpression(
+        data, "CTRL4", "DCs", DCCellGeneSet, rank, *ax[12:16]
+    )
+    plotDifferentialExpression(
+        data, "CTRL4", "pDCs", pDCGeneSet, rank, *ax[16:20], ct2=True,
+    )
+    plotDifferentialExpression(data, "CTRL4", "T Cells", TCellGeneSet, rank, *ax[20:24])
+    plotDifferentialExpression(
+        data, "CTRL4", "Cytotoxic T Cells", CytoTCellGeneSet, rank, *ax[24:28], ct2=True
+    )
+    plotDifferentialExpression(
+        data, "CTRL4", "Memory T Cells", memorytcellgeneset, rank, *ax[28:32], ct2=True
+    )
+    plotDifferentialExpression(
+        data, "CTRL4", "cDCs", cDCgeneset, rank, *ax[32:36], ct2=True
+    )
+    return f
 
-    origX = pf2(data, rank, doEmbedding=False)
 
-    plotConditionsFactors(
-        origX, ax[0], groupDrugs(origX.obs["Condition"]), ThomsonNorm=True
-    )
-    plotCellState(origX, ax[1])
-    plotGeneFactors(origX, ax[2])
+def plotDifferentialExpression(
+    data,
+    condition,
+    cell_type,
+    geneset,
+    rank,
+    *args,
+    ct2=False,
+    override=None
+):
+    sampled_data = None
+    if not ct2:
+        sampled_data = data[
+            (data.obs["Cell Type"] != cell_type) | (data.obs["Condition"] != condition)
+        ]
+    else:
+        sampled_data = data[
+            (data.obs["Cell Type2"] != cell_type) | (data.obs["Condition"] != condition)
+        ]
 
-    # sampled_data = data
+    origX = pf2(data, rank, doEmbedding=False)
 
     sampledX = pf2(sampled_data, rank, doEmbedding=False)
 
-    plotConditionsFactors(
-        sampledX, ax[3], groupDrugs(origX.obs["Condition"]), ThomsonNorm=True
-    )
-    plotCellState(sampledX, ax[4])
-    plotGeneFactors(sampledX, ax[5])
+    isolated_orig_vals = plotGeneFactorsIsolated(origX, args[0], geneset)
+    isolated_vals = plotGeneFactorsIsolated(sampledX, args[1], geneset)
+    most_exp_cmp = -1
+    most_exp = -1
+    most_exp_cmp2 = -1
+    most_exp2 = -1
 
-    return f
+    if override is None:
+        for i in range(isolated_orig_vals.shape[1]):
```

Comment:
Add comments for what these more complex blocks do. I can't follow, but there is likely a much simpler approach.